replica/database.cc

```diff
@@ -1324,23 +1328,29 @@ future<> database::create_in_memory_keyspace(const lw_shared_ptr<keyspace_metada
     }
     keyspace ks(ksm, std::move(kscfg), erm_factory);
     co_await ks.create_replication_strategy(get_shared_token_metadata());
-    _keyspaces.emplace(ksm->name(), std::move(ks));
+    co_return ks;
 }
 
-future<>
+future<std::unique_ptr<keyspace>>
```

Comment:
Is there some reason why `create_keyspace` returns a unique ptr and `create_in_memory_keyspace` doesn't?

Reply:
no, I will propagate unique_ptr there